### PR TITLE
Port changes from maven-shared

### PR DESCRIPTION
**Pull request by stevie400**
This ports the changes from https://github.com/apache/maven-shared/compare/trunk...HubSpot:hubspot. I did this by creating a patch file: 
```bash
➜  maven-shared git:(hubspot) git diff 622f1e81da78750c1a6054e09568ac6bdda4e9be -- maven-dependency-analyzer
```
And then applying it:
```bash
➜  maven-dependency-analyzer git:(port-changes) patch -p2 < ../maven-shared/changes.patch
```

@jhaber @kmclarnon

### CODE REVIEW

File: .hubspot-blazar-discovery.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+disabledDiscoveries: ["maven"]
```
Comment (jhaber) [inline .hubspot-blazar-discovery.yaml:1]: Now that it's not multi-module we probably don't need this

File: .blazar.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+before:
+  - description: "Change version to 1.11.2-hubspot-SNAPSHOT"
+    commands:
+      - sed -i 's~^  <version>1.11.2-hubspot-SNAPSHOT</version>$~  <version>1.11.2-hubspot-SNAPSHOT</version>~' pom.xml
+
+buildpack:
+  host: git.hubteam.com
+  organization: HubSpot
+  repository: Blazar-Buildpack-Java
+  branch: v2
```
Comment (jhaber) [inline .blazar.yaml:14]: Nor the buildpack section

File: .hubspot-blazar-discovery.yaml (hunk 1 above)
Comment (stevie400) [reply .hubspot-blazar-discovery.yaml:1]: We do because it discovers a bunch of test code.

File: .blazar.yaml (hunk 2 above)
Comment (stevie400) [reply .blazar.yaml:14]: Without the buildpack I see 
image

Any ideas?

Comment (jhaber) [reply .blazar.yaml:14]: You're right, removing the buildpack section was predicated upon removing `.hubspot-blazar-discovery.yaml`, but if we need one then we need the other

**Pull request merged**